planner/planner.go

```diff
@@ -3,8 +3,13 @@
 
 package planner
 
+import "go.jetpack.io/devbox/boxcli/usererr"
+
 type Planner interface {
 	Name() string
+	// IsBuildable returns true if the planner can build the project.
+	// It assumes that IsRelevant() has already returned true.
+	IsBuildable(srcDir string) (bool, error)
```

Comment:
Good idea, I can do this. I do think we need to check all stages are missing since some languages wont have a formal build stage (e.g. php does't have build maybe?)